manifests/baremetal/keepalived.yaml

```diff
@@ -49,26 +49,57 @@ spec:
         if pid=$(pgrep -o keepalived); then
             kill -s SIGHUP "$pid"
         else
-            /usr/sbin/keepalived -f /etc/keepalived/keepalived.conf --dont-fork --vrrp --log-detail --log-console &
+            start_keepalived
         fi
       }
+
+      stop_keepalived()
+      {
+        if pid=$(pgrep -o keepalived); then
+          kill -s SIGTERM "$pid"
+          # The monitor runs every 10 seconds
+          sleep 9
+          if pid=$(pgrep -o keepalived); then
+            kill -s SIGKILL "$pid"
+          fi
+          touch /var/run/keepalived/stopped
+        fi
+      }
+
+      start_keepalived()
+      {
+        if ! pgrep -o keepalived > /dev/null; then
+          /usr/sbin/keepalived -f /etc/keepalived/keepalived.conf --dont-fork --vrrp --log-detail --log-console &
+        fi
+        rm -f /var/run/keepalived/stopped
+      }
+
       msg_handler()
       {
         while read -r line; do
           echo "The client sent: $line" >&2
-          # currently only 'reload' msg is supported
           if [ "$line" = reload ]; then
               reload_keepalived
+          elif [ "$line" = stop ]; then
+              stop_keepalived
+          elif [ "$line" = start ]; then
+              start_keepalived
+          else
+              echo "Unrecognized command: $line" >&2
           fi
         done
       }
+
       set -ex
       declare -r keepalived_sock="/var/run/keepalived/keepalived.sock"
       export -f msg_handler
       export -f reload_keepalived
+      export -f stop_keepalived
+      export -f start_keepalived
       if [ -s "/etc/keepalived/keepalived.conf" ]; then
```

Comment:
Do we want to cover also the following (corner) case ? : 
1. Monitor container send STOP request 
   1.1 Keepalived container stop Keepalived process
2. After sometime Kubelet restart Kaapalived container for some reason

Reply:
I'm inclined to say no. If the whole keepalived container gets restarted, I think we should let it attempt to run normally. If the monitor continues to fail it will stop it again anyway.